src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -60,16 +60,19 @@ public EiffelEventSender(EiffelPluginConfiguration pluginConfig, HttpRequest htt
     }
 
     /**
-     * Sends a REMReM eiffel message using
+     * Sends a REMReM Eiffel message to the generateAndPublish endpoint. RuntimeException is thrown
+     * when a HttpRequestFailedException occurs so that the retry logic works.
      *
      */
     public void send() {
         try {
             verifyConfiguration();
             generateAndPublish();
-        } catch (URISyntaxException | IOException | MissingConfigurationException
-                | HttpRequestFailedException e) {
-            LOGGER.error("Failed to send eiffel message.", e);
+        } catch (URISyntaxException | IOException | MissingConfigurationException e1) {
+            LOGGER.error("Failed to send eiffel message.", e1);
+        } catch (HttpRequestFailedException e2) {
+            LOGGER.error("Failed to send eiffel message.", e2);
+            throw e2;
```

Comment:
I think there are two flaws with this logic:

* We're unconditionally retrying all non-200 HTTP codes, including various 4xx errors that aren't likely to get fixed with us retrying (e.g. 400, 405, and 409).
* IOException errors should in many (most?) cases be retried since they could very well be the result of ETIMEDOUT, ECONNREFUSED, or EHOSTUNREACH errors, or temporary DNS lookup problems.

Reply:
Yes, previously when I played around the apache retry logic I had more code checks, but this was just implemented on top of the current code without further changes to the Exception handling. There is also 401 where you might not want to retry if authorization fails etc. I will look more into this and consult my colleagues.

Reply:
Indeed, 401 shouldn't be retried (hello, instant account lockout) but one could make the argument that 403 errors _should_ be retried to better cope with temporary misconfigurations.